### PR TITLE
Changes related to creation of a modular backend in multiprecision [SyncWith: crypto3-multiprecision#70] 

### PR DESCRIPTION
**Pull request by martun**

### CODE REVIEW

File: .github/workflows/publish-results.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ on:
 jobs:
   call-reusable-workflow:
     name: Call Reusable Testing Callback Workflow
-    uses: NilFoundation/ci-cd/.github/workflows/reusable-crypto3-publish-result.yml@v1
+    uses: NilFoundation/ci-cd/.github/workflows/reusable-crypto3-publish-result.yml@v1.2.0
```
Comment (x-mass) [inline .github/workflows/publish-results.yml]: 1.2.1

File: .github/workflows/pull-request.yml
Comment (x-mass) [inline .github/workflows/pull-request.yml:1]: Why did you introduce these changes? Tests were failing on mac?

We could use nix testing pipelines alredy, but feel free to leave to to DevOps :3

File: .github/workflows/set_version.yml
Comment (x-mass) [inline .github/workflows/set_version.yml:1]: I'm not sure if we need such workflows in our repos

**Review by x-mass — APPROVED** [review]

File: .github/workflows/publish-results.yml (hunk 1 above)
Comment (martun) [reply .github/workflows/publish-results.yml]: changed.

File: .github/workflows/set_version.yml
Comment (martun) [reply .github/workflows/set_version.yml:1]: Removed.

File: .github/workflows/pull-request.yml
Comment (martun) [reply .github/workflows/pull-request.yml:1]: Removed the list of targets and the commented part of mac.

**Pull request merged**